aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1897,34 +1855,18 @@ impl AptosVM {
             arguments,
             &mut gas_meter,
         ) {
-            Ok(result) => {
-                ViewFunctionOutput::new(Ok(result), Self::gas_used(gas_budget, &gas_meter))
+            Ok(result) => ViewFunctionOutput::new(
+                Ok(result),
+                Self::gas_used(max_gas_amount.into(), &gas_meter),
+            ),
+            Err(e) => {
+                ViewFunctionOutput::new(Err(e), Self::gas_used(max_gas_amount.into(), &gas_meter))
             },
-            Err(e) => ViewFunctionOutput::new(Err(e), Self::gas_used(gas_budget, &gas_meter)),
         }
     }
 
-    fn memory_tracked_gas_meter(
-        vm: &AptosVM,
-        log_context: &AdapterLogSchema,
-        gas_budget: u64,
-    ) -> Result<MemoryTrackedGasMeter<StandardGasMeter<StandardGasAlgebra>>> {
-        let gas_meter = MemoryTrackedGasMeter::new(StandardGasMeter::new(StandardGasAlgebra::new(
-            vm.gas_feature_version,
-            get_or_vm_startup_failure(&vm.gas_params, log_context)?
-                .vm
-                .clone(),
-            get_or_vm_startup_failure(&vm.storage_gas_params, log_context)?.clone(),
-            gas_budget,
-        )));
-        Ok(gas_meter)
-    }
-
-    fn gas_used(
-        gas_budget: u64,
-        gas_meter: &MemoryTrackedGasMeter<StandardGasMeter<StandardGasAlgebra>>,
-    ) -> u64 {
-        GasQuantity::new(gas_budget)
+    fn gas_used(max_gas_amount: Gas, gas_meter: &impl AptosGasMeter) -> u64 {
```

Comment:
lol, let's start unit testing aptos_vm by should panic and not should panic unit tests? (if having mock gas meter isn't too difficult?)

Reply:
Good point, right now there are no tests for VM apart from those that test other features and work as expected lol